tests/utils.py

```diff
@@ -21,6 +22,102 @@
 from m2cgen import ast
 
 
+class StatsmodelsSklearnLikeWrapper(BaseEstimator, RegressorMixin):
+    def __init__(self, model, params):
+        self.model = model
+        self.params = params
+        self.__class__.__name__ = model.__name__
+
+    def fit(self, X, y):
+        init_params = self.params.get("init", {})
+        self.fit_intercept_ = init_params.pop("fit_intercept", False)
+        if self.fit_intercept_:
+            X = sm.add_constant(X)
+        est = self.model(y, X, **init_params)
+        if "fit_regularized" in self.params:
+            self.fitted_model_ = est.fit_regularized(
+                **self.params["fit_regularized"])
+        else:
+            self.fitted_model_ = est.fit(**self.params.get("fit", {}))
+        self.__class__.__name__ = type(self.fitted_model_).__name__
+        return self.fitted_model_
+
+    def predict(self, X):
+        if self.fit_intercept_:
+            X = sm.add_constant(X)
+        return self.fitted_model_.predict(X)
+
+
+class ModelTrainer:
+
+    _class_instances = {}
+
+    def __init__(self, dataset_name, test_fraction):
+        self.dataset_name = dataset_name
+        self.test_fraction = test_fraction
+        np.random.seed(seed=7)
+        if dataset_name == "boston":
+            self.__name__ = "train_model_regression"
```

Comment:
Seems more like a workaround to me. Can we make it an actual `@property` like `dataset_name` instead? I assume it's only used in a single place here: https://github.com/BayesWitnesses/m2cgen/blob/master/tests/utils.py#L191. We just need to use `trainer.dataset_name` instead of the `trainer.__name__`

Reply:
Sure! Good catch! 👍